StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/Link/PaymentDetails.swift

```diff
@@ -140,82 +103,45 @@ extension ConsumerPaymentDetails.Details {
             case unknown = "UNKNOWN"
         }
 
-        let cvcCheck: State
-
-        var allResponseFields: [AnyHashable: Any]
+        let cvcCheck: CVCCheck
 
-        init(cvcCheck: State, allResponseFields: [AnyHashable: Any]) {
+        init(cvcCheck: CVCCheck) {
             self.cvcCheck = cvcCheck
-            self.allResponseFields = allResponseFields
-            super.init()
-        }
-
-        static func decodedObject(fromAPIResponse response: [AnyHashable: Any]?) -> Self? {
-            guard
-                let dict = response,
-                let cvcCheck = dict["cvc_check"] as? String
-            else {
-                return nil
-            }
-
-            let cvcCheckState = State(rawValue: cvcCheck.uppercased()) ?? .unknown
-
-            return CardChecks(
-                cvcCheck: cvcCheckState,
-                allResponseFields: dict) as? Self
         }
     }
 }
 
 // MARK: - Details.Card
 extension ConsumerPaymentDetails.Details {
-    class Card: NSObject, STPAPIResponseDecodable {
-
+    final class Card: Codable {
         let expiryYear: Int
         let expiryMonth: Int
-        let brand: STPCardBrand
+        let brand: String
         let last4: String
         let checks: CardChecks?
 
-        let allResponseFields: [AnyHashable: Any]
+        private enum CodingKeys: String, CodingKey {
```

Comment:
Are these coming from the service as `expYear` or `exp_year`? If you use the camelCase version of the server representation (`exp_year` or `expYear`), you don't have to implement CodingKeys at all.

Reply:
`exp_year`, though a bit selfishly I prefer the unabbreviated version, hence the mapping 😅 Probably not worth it just to have a slightly nicer Swift API, though!

Reply:
Meh, up to you! Explicitly defining the `CodingKeys` can make the code a little easier to understand at the expense of a little more boilerplate.